engine/engine.go

```diff
@@ -129,26 +136,38 @@ func New(opts Opts) *compatibilityEngine {
 		level.Debug(opts.Logger).Log("msg", "lookback delta is zero, setting to default value", "value", 5*time.Minute)
 	}
 
-	return &compatibilityEngine{
-		prom: promql.NewEngine(opts.EngineOpts),
+	metrics := &engineMetrics{
+		currentQueries: promauto.With(opts.Reg).NewGauge(
+			prometheus.GaugeOpts{
+				Namespace: namespace,
+				Subsystem: subsystem,
+				Name:      "queries",
+				Help:      "The current number of queries being executed or waiting.",
+			},
+		),
 		queries: promauto.With(opts.Reg).NewCounterVec(
 			prometheus.CounterOpts{
 				Name: "promql_engine_queries_total",
```

Comment:
I wonder if the name should be changed to `thanos_engine_queries_total` as well? Wdyt @GiedriusS?

Reply:
Makes sense, we can re-use the same pattern as in `currentQueries` :+1: